client/mifare4.c

```diff
@@ -16,6 +16,18 @@
 #include "ui.h"
 #include "polarssl/libpcrypto.h"
 
+int CalulateMAC(mf4Session *session, uint8_t *data, int datalen, uint8_t *mac, bool verbose) {
+	if (!session || !session->Authenticated || !mac || !data || !datalen)
+		return 1;
+	
+	memset(mac, 0x00, 8);
+	
+	if (verbose)
+		PrintAndLog("MAC data[%d]: %s", datalen, sprint_hex(data, datalen));
+	
```

Comment:
see http://www.proxmark.org/forum/viewtopic.php?id=5348 for the input to CMAC calculation.

Reply:
cool)